plugin/PlayAssetDelivery/src/main/java/com/google/play/core/godot/assetpacks/utils/AssetPackStatesFromDictionary.java

```diff
@@ -0,0 +1,73 @@
+/*
+ *  	Copyright 2020 Google LLC
+ *
+ *  	Licensed under the Apache License, Version 2.0 (the "License");
+ *  	you may not use this file except in compliance with the License.
+ *  	You may obtain a copy of the License at
+ *
+ *  		https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  	Unless required by applicable law or agreed to in writing, software
+ *  	distributed under the License is distributed on an "AS IS" BASIS,
+ *  	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  	See the License for the specific language governing permissions and
+ *  	limitations under the License.
+ */
+
+package com.google.play.core.godot.assetpacks.utils;
+
+import com.google.android.play.core.assetpacks.AssetPackState;
+import com.google.android.play.core.assetpacks.AssetPackStates;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.godotengine.godot.Dictionary;
+
+/**
+ * This class extends the AssetPackStates abstract class, and provides constructor that allows
+ * PlayAssetDeliveryUtils class to instantiate AssetPackStatesFromDictionary given a Godot
+ * Dictionary.
+ */
+public class AssetPackStatesFromDictionary extends AssetPackStates {
+  private long totalBytes;
+  private Map<String, AssetPackState> packStates;
+
+  public static final String TOTAL_BYTES_KEY = "totalBytes";
+  public static final String PACK_STATES_KEY = "packStates";
+
+  private static final Set<String> dictionaryRequiredKeySet =
+      new HashSet<>(Arrays.asList(TOTAL_BYTES_KEY, PACK_STATES_KEY));
+
+  public AssetPackStatesFromDictionary(Dictionary dict) throws IllegalArgumentException {
+    if (dict.keySet().containsAll(dictionaryRequiredKeySet)) {
+      try {
+        this.totalBytes = (long) dict.get(TOTAL_BYTES_KEY);
+        Dictionary packStatesDictionary = (Dictionary) dict.get(PACK_STATES_KEY);
+        Map<String, AssetPackState> returnMap = new HashMap<String, AssetPackState>();
+        for (Map.Entry<String, Object> entry : packStatesDictionary.entrySet()) {
```

Comment:
You could use streams to make this a little more concise:
```
this.packStates = packStatesDictionary.entrySet().stream().collect(Collectors.ToMap( entry -> entry.getKey(), entry -> new AssetPackStateFromDictionary((Dictionary) entry.getValue())))
```

It also clarifies that you're converting one dictionary to another and eliminates the need for the intermediate variable (`returnMap`).

Reply:
I looked into integrating the streams approach. However, the current ```minSdkVersion``` version for the plugin is 18, which will result in a compile-time error since stream requires API level 24+.
This ```minSdkVersion``` of 18, also used by the [GodotPaymentPlugin](https://github.com/godotengine/godot-google-play-billing/blob/916385d3ca2dbbff493fb53272921b1ca8d71e65/godot-google-play-billing/build.gradle#L13) and [GodotOculusMobilePlugin](https://github.com/GodotVR/godot_oculus_mobile/blob/aadaf1bfb04eb58bd3ea11dedab75253943d835f/config.gradle.kts#L6), is currently required for Android plugins. I have tried configuring build.gradle with higher API level and will result in an error when I export the Godot game.
I think it is possible to bump the ```minSdkVersion``` version by recompiling the engine, but that might be too costly and I am not sure about the side effects. I'm just wondering is there any work arounds for this situation? Or is it acceptable to keep the current entrySet approach?

Reply:
Adapted the stream approach.